docs/measuring_performance.md

```diff
@@ -33,3 +33,67 @@ The time for each action is split into at 3 primary pieces:
 If the builder uses a `Resolver`, you will also see a breakdown of time spent
 getting the resolver. This will appear below the `build` time since it overlaps
 with it.
+
+### Slices
+
+A "slice" is a piece of actual synchronous work, up to the granularity of the
+[slices resolution](#slices-resolution) configuration, visualized as a
+horizontal bar in a row in the graph. It does not include asynchronous work
+(unless it is less than the slice resolution).
+
+If [show async slices](#show-async-slices) is disabled, the granularity is
+infinite, so a slice is just the entire time spent in a stage, including all
+async work.
+
+### Stage time, slice time, user time, and real time
+
+Within each action (row), you can hover over one of the active time slices,
+which will show you three several metrics, described as follows:
+
+**Stage time**: The start and end time (relative to the start of the build) for
+the stage of the time slice you are currently hovering over.
+
+**Slice time**: The start and end time (relative to the start of the build) for
+the slice.
+
+**Slice duration**: The total time spent on the current slice.
+
+**User time**: The actual time spent synchronously performing actions related to
+the current stage. This does not count time spent waiting for asynchronous tasks.
+This is generally the most relevant time, because asynchronous reads of assets
+as an example may require actually building those files.
+
+**Real time**: The total time between the start and end time of this stage. This
+includes time waiting for asynchronous tasks.
+
+## Configuring the timeline
+
+### Hide skipped actions
+
+Many actions in a build are "optional", which means they only run if some other
+action asks to read their output. This setting toggles whether those rows
```

Comment:
Double checking - is skipped only the optional assets, or does it include not-invalidated assets?